### PR TITLE
Fix 5641 - SceneSystem defined as a data provider

### PR DESCRIPTION
**Pull request by davidkline-ms**
IMixedRealitySceneSystem was defines as both a CoreSytem (inherits BaseCoreSystem) and an IMixedRealityDataProvider.

This is incorrect and was preventing the scene service from being loadable.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystem.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace Microsoft.MixedReality.Toolkit.SceneSystem
     /// This part handles the runtime parts of the service.
     /// </summary>
     [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/SceneSystem/SceneSystemGettingStarted.html")]
-    public partial class MixedRealitySceneSystem : BaseCoreSystem, IMixedRealitySceneSystem
+    public partial class MixedRealitySceneSystem : BaseCoreSystem
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystem.cs]: Shouldn't this stay the same?

```suggestion
    public partial class MixedRealitySceneSystem : BaseCoreSystem, IMixedRealitySceneSystem
```

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystem.cs]: Unneeded. BaseCoreSystem already specifies IMixedRealityService

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystem.cs]: Good catch! My bad reading.. :D

**Comment by wiwei** [comment]
We should absolutely have a test for this so that things don't error like this going forward.

**Comment by davidkline-ms** [comment]
> We should absolutely have a test for this so that things don't error like this going forward.

Agreed. This will need to be done for _all_ core services and MRTK team provided extension services.

**Comment by Troy-Ferrell** [comment]
> We should absolutely have a test for this so that things don't error like this going forward.
> 
> Agreed. This will need to be done for all core services and MRTK team provided extension services.
> Not sure how to make a catch-all for this though

I can look at possibly including with #5579

**Pull request merged**